src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -322,8 +322,28 @@ private async Task MoveSegmentsToProject(string botSegmentsDirectoryPrefix)
                 // So we can safely create this without interfering with the .Move command right below.
                 Directory.CreateDirectory(Path.GetDirectoryName(segmentResourceDirectory));
 
-                // move the directory (this also deletes the source directory)
-                Directory.Move(botSegmentsDirectoryPrefix, segmentResourceDirectory);
+                // If on a Mac, for some weird reason, .Move will fail for project paths that have spaces. We predict
+                // that downstream, something is happening where the space is not properly handled in Unix commands. 
+                // Unfortunately, we could only reproduce this once, but this change should fix it. On Mac, we instead
+                // read and write the files by hand, and avoid any move operations.
+                if (Application.platform == RuntimePlatform.OSXEditor || Application.platform == RuntimePlatform.OSXPlayer)
```

Comment:
You should only need OSXPlayer as this code is run in the context of the game playing, not an editor script

Reply:
Ah thank you

Reply:
@RG-nAmKcAz After some testing, when running in the editor as the game is playing, the platform is indeed `RuntimePlatform.OSXEditor`, not `RuntimePlatform.OSXPlayer`, so I have to keep this here. It is OSXEditor even if not an editor script.